src/main/java/stirling/software/SPDF/config/security/mail/MailConfig.java

```diff
@@ -0,0 +1,52 @@
+package stirling.software.SPDF.config.security.mail;
+
+import java.util.Properties;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import stirling.software.SPDF.model.ApplicationProperties;
+
+/**
+ * This configuration class provides the JavaMailSender bean, which is used to send emails. It reads
+ * email server settings from the configuration (ApplicationProperties) and configures the mail
+ * client (JavaMailSender).
+ */
+@Configuration
+@Slf4j
+@AllArgsConstructor
+public class MailConfig {
+
+    private final ApplicationProperties applicationProperties;
+
+    @Bean
```

Comment:
Shall we make this conditional to the property being present?
To avoid needless bean creation

Reply:
Good point, I'll add an option `ConditionalOnProperty`